eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/common/BaseBackStackedRetainedFragment.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.common;
+
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+
+/**
+ * Base class for retained Fragments that may be popped back from Fragment back stack in the app.
+ * It provides and extra boolean flag in onCreateView() for that scenario.
+ *
+ * @author Gabor Keszthelyi
+ */
+public abstract class BaseBackStackedRetainedFragment extends BaseFragment
+{
+    private boolean mIsInitializing = true;
+
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState)
+    {
+        super.onCreate(savedInstanceState);
+        setRetainInstance(true);
+    }
+
+
+    @Nullable
+    @Override
+    public final View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState)
+    {
+        View view = onCreateView(inflater, container, savedInstanceState, mIsInitializing);
```

Comment:
I'm not a big fan of the template method pattern.
Also, I consider this only a temporary workaround due to the lack of a better backstack implementation. No other Fragment we create should need this, so think it's not worth creating a separate abstraction class for this. That only encourages using this more often when we really should search for a better solution instead.